test/index.js

```diff
@@ -460,4 +470,124 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('cacheEncryptionKey', function () {
+    it('sets encryption key data upon submitPassword', async function () {
+      keyringController.cacheEncryptionKey = true;
+      await keyringController.submitPassword(password);
+
+      expect(keyringController.password).toBe(password);
+      expect(keyringController.memStore.getState().encryptionSalt).toBe('SALT');
+      expect(keyringController.memStore.getState().encryptionKey).toStrictEqual(
+        expect.stringMatching('.+'),
+      );
+    });
+
+    it('unlocks the keyrings with valid information', async function () {
+      keyringController.cacheEncryptionKey = true;
+      const returnValue = await keyringController.encryptor.decryptWithKey();
+      const stub = sinon.stub(keyringController.encryptor, 'decryptWithKey');
```

Comment:
Nit: IMO this would be a better test with this removed. The keyrings-based assertion is enough to show this works; this part just makes the test more fragile by tying it to internals.